packages/block-directory/src/store/actions.js

```diff
@@ -45,88 +50,44 @@ export function setInstallBlocksPermission( hasPermission ) {
 }
 
 /**
- * Action triggered to download block assets.
+ * Action triggered to install a block plugin.
  *
- * @param {Object} item The selected block item
- * @param {Function} onSuccess The callback function when the action has succeeded.
- * @param {Function} onError The callback function when the action has failed.
+ * @param {Object} item The block item returned by search.
  */
-export function* downloadBlock( item, onSuccess, onError ) {
+export function* installBlockType( { id, name, assets } ) {
+	yield clearErrorNotice( id );
 	try {
-		if ( ! item.assets.length ) {
+		if ( ! Array.isArray( assets ) || ! assets.length ) {
 			throw new Error( 'Block has no assets' );
 		}
-
-		yield loadAssets( item.assets );
-		const registeredBlocks = getBlockTypes();
-		if ( registeredBlocks.length ) {
-			onSuccess( item );
-		} else {
-			throw new Error( 'Unable to get block types' );
-		}
-	} catch ( error ) {
-		yield onError( error );
-	}
-}
-
-/**
- * Action triggered to install a block plugin.
- *
- * @param {string} item The block item returned by search.
- * @param {Function} onSuccess The callback function when the action has succeeded.
- * @param {Function} onError The callback function when the action has failed.
- *
- */
-export function* installBlock( { id, name }, onSuccess, onError ) {
-	try {
+		yield setIsInstalling( true );
 		const response = yield apiFetch( {
 			path: '__experimental/block-directory/install',
 			data: {
 				slug: id,
 			},
 			method: 'POST',
 		} );
-		if ( response.success === false ) {
+		if ( response.success !== true ) {
 			throw new Error( response.errorMessage );
```

Comment:
I tested locally, if the length of this error message is `< 1`, no error message will be displayed.

Reply:
I wonder if we should consider a default error message?

Reply:
Added the oh-so-descriptive "An error occurred." as a default. Hopefully we'll have error content for most detectable errors, so no one will see this 🙂

Reply:
If an error occurs here it'd be a proper REST API error response. AFAIK, that throws. I don't think this code path could get executed. Am I missing something?

Reply:
Yeah, this would probably never be executed (and if it is, errorMessage wouldn't exist) — but it could if the response from the server was malformed somehow, or the API endpoint could change to return failure info. I've updated it to show a default message, and we can look at removing it if needed when the API settles some.